src/shared/lib/table.ts

```diff
@@ -78,3 +79,67 @@ export function useTable<T>(fetchFn: TableFetchFn<T>) {
     setSize,
   };
 }
+// for generating blocksListTable in paginated way
+export function useBlockTable(selectedTransaction: TransactionDto[]) {
```

Comment:
`shared/lib/table` contains _shared_ utils related to tables all across the application. `useBlockTable` should be placed somewhere in the scope of the block details page/feature. You need to comprehend how [Feature-Sliced Design](https://feature-sliced.design/) works.

Apart from that, `useBlockTable` seems to be almost a full copy of `useTable`.